dialer.go

```diff
@@ -571,13 +571,21 @@ func (d *Dialer) Warmup(ctx context.Context, icn string, opts ...DialOption) err
 // newInstrumentedConn initializes an instrumentedConn that on closing will
 // decrement the number of open connects and record the result.
 func newInstrumentedConn(conn net.Conn, closeFunc func(), errFunc func(error), dialerID, connName string) *instrumentedConn {
-	return &instrumentedConn{
+	reporterCtx, stopReporter := context.WithCancel(context.Background())
```

Comment:
ctx, cancel would be fine here.

https://google.github.io/styleguide/go/decisions#variable-names